staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -223,6 +223,8 @@ func (h *UpgradeAwareHandler) ServeHTTP(w http.ResponseWriter, req *http.Request
 	newReq.Header = utilnet.CloneHeader(req.Header)
 	if !h.UseRequestLocation {
 		newReq.URL = &loc
+		newReq.Host = loc.Host
+		newReq.RequestURI = loc.Path
```

Comment:
this really needs a test demonstrating the issue this resolves and preventing regression